src/Microsoft.Data/DecimalConverter.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Data
+{
+    internal interface IDecimalConverter<T>
+        where T : struct
+    {
+        decimal GetDecimal(T value);
+    }
+    internal static class DecimalConverter<T>
+        where T : struct
+    {
+        public static IDecimalConverter<T> Instance => DecimalConverter.GetDecimalConverter<T>();
```

Comment:
This and DoubleConverter suffer from the same pattern that `PrimitiveColumnArithmetic` did - every time you invoke this property it allocates a new object.

Reply:
Fixed